components/org.wso2.identity.event.http.publisher/src/main/java/org/wso2/identity/event/http/publisher/internal/service/impl/HTTPEventPublisherImpl.java

```diff
@@ -86,63 +82,101 @@ public boolean canHandleEvent(EventContext eventContext) throws EventPublisherEx
+                                HTTPAdapterConstants.LogConstants.ActionIDs.PUBLISH_EVENT,
+                                DiagnosticLog.ResultStatus.FAILED,
+                                "Publish attempt failed due to exception. Retrying… (" +
+                                        retriesLeft + " attempts left)");
+                        sendWithRetries(eventPayload, eventContext, url, secret, retriesLeft - 1);
+                    } else {
                         handleResponseCorrelationLog(request, requestStartTime,
                                 HTTPCorrelationLogUtils.RequestStatus.FAILED.getStatus(),
-                                ex.getMessage());
+                                throwable.getMessage());
+                        printPublisherDiagnosticLog(eventContext, eventPayload, url,
+                                HTTPAdapterConstants.LogConstants.ActionIDs.PUBLISH_EVENT,
+                                DiagnosticLog.ResultStatus.FAILED,
+                                "Failed to publish event data to endpoint.");
                         log.warn("Failed to publish event data to endpoint: " + url);
-                        log.debug("Failed to publish event data to endpoint. ", ex);
-                        return null;
-                    });
-            log.debug("Event published successfully to the HTTP. Endpoint: " + url);
-        }
-    }
-
-    private static void handleAsyncResponse(HttpResponse response, HttpPost request, long requestStartTime,
-                                            EventContext eventContext, String url, String eventUri) {
-
-        PrivilegedCarbonContext.startTenantFlow();
-        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(eventContext.getTenantDomain());
-
-        try {
-            int responseCode = response.getStatusLine().getStatusCode();
-            String responsePhrase = response.getStatusLine().getReasonPhrase();
-            log.debug("HTTP request completed. Response code: " + responseCode +
-                    ", Endpoint: " + url + ", Event URI: " + eventUri);
-
-            handleResponseCorrelationLog(request, requestStartTime,
-                    HTTPCorrelationLogUtils.RequestStatus.COMPLETED.getStatus(),
-                    String.valueOf(responseCode), responsePhrase);
-            if (responseCode >= 200 && responseCode < 300) {
-                logDiagnosticSuccess(eventContext, url, eventUri);
-                log.debug("HTTP request completed. Response code: " + responseCode +
-                        ", Endpoint: " + url + ", Event URI: " + eventUri);
-            } else {
-                logDiagnosticFailure(eventContext, url, eventUri);
-                log.debug("HTTP request failed with response code: " + responseCode +
-                        " due to: " + responsePhrase + ". Endpoint: " + url);
+                        log.debug("Failed to publish event data to endpoint. ", throwable);
```

Comment:
Shall we add `url ` here to this log as well ?

Reply:
Check other occurrences as well.

Reply:
Fixed via https://github.com/wso2-extensions/identity-event-publishers/pull/41/commits/3fa6b2d6efa1dee4fb2fe934d507300b04416fbf